lib/Backend/GlobOpt.cpp

```diff
@@ -9230,47 +9243,77 @@ GlobOpt::OptConstFoldBranch(IR::Instr *instr, Value *src1Val, Value*src2Val, Val
     case Js::OpCode::BrNotNeq_A:
         if (!src1Var || !src2Var)
         {
-            return false;
+            ValueInfo *src1ValInfo = src1Val->GetValueInfo();
+            ValueInfo *src2ValInfo = src2Val->GetValueInfo();
+            if (
+                (src1ValInfo->IsNumber() && src1Var && src2ValInfo->IsBoolean() && src1Var != Js::TaggedInt::ToVarUnchecked(0) && src1Var != Js::TaggedInt::ToVarUnchecked(1)) ||
```

Comment:
Can we also fold a branch when we get a non-null constant var for the src which is definitely boolean and then compare the constant boolean with the constant number value of the other src?